packages/compass-e2e-tests/helpers/smoke-test/build-info.ts

```diff
@@ -0,0 +1,189 @@
+  assertObjectHasKeys(buildInfo, 'buildInfo', rhelFilenameKeys);
+}
+
+export type PackageDetails = {
+  kind: PackageKind;
+  filename: string;
+} & (
+  | {
+      kind: 'windows_setup' | 'windows_msi' | 'windows_zip';
+      buildInfo: WindowsBuildInfo;
+    }
+  | {
+      kind: 'osx_dmg' | 'osx_zip';
+      buildInfo: OSXBuildInfo;
+    }
+  | {
+      kind: 'linux_deb' | 'linux_tar';
+      buildInfo: UbuntuBuildInfo;
+    }
+  | {
+      kind: 'linux_rpm' | 'rhel_tar';
+      buildInfo: RHELBuildInfo;
+    }
+);
+
+/**
+ * Extracts the filename of the packaged app from the build info, specific to a kind of package.
+ */
+export function getPackageDetails(
+  kind: PackageKind,
+  buildInfo: unknown
+): PackageDetails {
+  if (
+    kind === 'windows_setup' ||
+    kind === 'windows_msi' ||
+    kind === 'windows_zip'
+  ) {
+    assertBuildInfoIsWindows(buildInfo);
+    return { kind, buildInfo, filename: buildInfo[`${kind}_filename`] };
+  } else if (kind === 'osx_dmg' || kind === 'osx_zip') {
+    assertBuildInfoIsOSX(buildInfo);
+    return { kind, buildInfo, filename: buildInfo[`${kind}_filename`] };
+  } else if (kind === 'linux_deb' || kind === 'linux_tar') {
+    assertBuildInfoIsUbuntu(buildInfo);
+    return { kind, buildInfo, filename: buildInfo[`${kind}_filename`] };
+  } else if (kind === 'linux_rpm' || kind === 'rhel_tar') {
+    assertBuildInfoIsRHEL(buildInfo);
+    return { kind, buildInfo, filename: buildInfo[`${kind}_filename`] };
+  } else {
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+    throw new Error(`Unsupported package kind: ${kind}`);
+  }
+}
+
+function readJson<T extends object>(...segments: string[]): T {
+  const result = JSON.parse(
+    fs.readFileSync(path.join(...segments), 'utf8')
+  ) as unknown;
+  assert(typeof result === 'object' && result !== null, 'Expected an object');
+  return result as T;
+}
+
+export function readPackageDetails(
+  kind: PackageKind,
+  filePath: string
+): PackageDetails {
+  const result = readJson(filePath);
+  return getPackageDetails(kind, result);
+}
+
+export function writeAndReadPackageDetails(
+  context: SmokeTestsContext
+): PackageDetails {
+  const compassDir = path.resolve(__dirname, '../../../compass');
+  const infoArgs = {
+    format: 'json',
+    dir: compassDir,
+    platform: context.platform,
+    arch: context.arch,
+    out: path.resolve(context.sandboxPath, 'target.json'),
+  };
+  console.log({ infoArgs });
+
+  // These are known environment variables that will affect the way
+  // writeBuildInfo works. Log them as a reminder and for our own sanity
+  console.log(
+    'info env vars',
+    pick(process.env, [
+      'HADRON_DISTRIBUTION',
+      'HADRON_APP_VERSION',
+      'HADRON_PRODUCT',
+      'HADRON_PRODUCT_NAME',
+      'HADRON_READONLY',
+      'HADRON_ISOLATED',
+      'DEV_VERSION_IDENTIFIER',
+      'IS_RHEL',
+    ])
+  );
+  writeBuildInfo(infoArgs);
```

Comment:
Part of me wants to change hadron-build so we can get this info as a variable without an intermediate file, but I've been a bit weary of changing that code because it is used in so many ways and often only at release time. Although I suppose nowadays we release dev too, so it should be safer.